### PR TITLE
[solana] Start to implement the solana receiver contract

### PR DESCRIPTION
**Pull request by yhzhang0128**

### CODE REVIEW

File: target_chains/solana/Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
...
...i post-price-vaa -v AQAAAAABAGL2NH+QGX4mcTw+F2uFuH8lC62485K1o/vxftbGKL1gauMUW6Lzb18JJLF4SJtkc9H5MnqC+Mpai+bDoj2Ztw8BY9p6/AAAAAAAAfNGGVrALzfWDU24/6bvdMsb41UAR1Q6Sp7prPTXhpewAAAAAAeb7UQBUDJXSAADAAAAAQIABQCVQxzC/Q70r0vHyF//ri9j1Rsm0WIXloLRSa5hmxIhwAv8MJRn3vpLGYxrW9WcCNtLnfsn3bzDLzFWDyF7T/j8KwAAACRAHYigAAAAAAPYMSD////4AAAAJCpaeTgAAAAAAqjStgEAAAABAAAAFQAAAABj2nr8AAAAAGPaevwAAAAAY9p6/AAAACRAHYigAAAAAAPYMSD0Kq+ITHsUVIlBcL4Krx2zm0t406VqJ/1JvYs57ywz12UQcfjHqyMhtr3TvHm5SlCEGpKm4GX547i5kmqPtaXRAAAAJO/rVQAAAAAABQG9AP////gAAAAk30Wg6AAAAAAFg7x3AQAAAAEAAAALAAAAAGPaevwAAAAAY9p6/AAAAABj2nr8AAAAJO/rVQAAAAAABQG9ABgB6wOAOvAkRSPuKobD8nsSar6JBNtLRagq21/iFwi0yoC6bcMuCNBvGqiGAR7tHXfHe+nrdhzBDXK30KL9V6YAAAAk8MYnegAAAAAD1cL6////+AAAACTf8vQAAAAAAANX/dkBAAAAAgAAAB4AAAAAY9p6/AAAAABj2nr8AAAAAGPaevwAAAAk8MYnegAAAAAD1cL6fd8Ngq9THwrxCdXpzp7Ce6nwDp7oq3HJEq//oW1xWDa3q9Jadt2v/fhHIk8DGYzLknI/kLJCnPM/Duy5bjUqhgAAACRKX0MAAAAAAIiMq2D////4AAAAJGpZPYAAAAAAf92Z9AEAAAACAAAADgAAAABj2nr8AAAAAGPaevwAAAAAY9p6/AAAACRKX0MAAAAAAIP9X0DVpcLzDga9bzjgHCxMjN18p8HBLUenM25Fn8bbQXG65mD9YbLZDrpH8oFQWoiGm2YTPZ3FjyA7AZ9apH8bOTQ+AAAAAAAAAAAAAAAAAAAAAP////gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALAAAAAGPaevwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==
```
Comment (guibescos) [inline target_chains/solana/Makefile:4]: I use `cargo run post-price-vaa` which seems better than relying on debug being populated

File: target_chains/solana/cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "solana-receiver-cli"
```
Comment (guibescos) [inline target_chains/solana/cli/Cargo.toml:2]: Can you put Pyth in the name. Changing the name later is hard.

File: target_chains/solana/programs/solana-receiver/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "solana-receiver"
```
Comment (guibescos) [inline target_chains/solana/programs/solana-receiver/Cargo.toml:2]: I think Pyth should be in the name. Changing the name later is hard.

File: target_chains/solana/cli/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            // Hard-coded strings
+            let rpc_client = RpcClient::new("https://api.devnet.solana.com");
+            // Is RpcClient::new_with_commitment necessary?
+            let wormhole = Pubkey::from_str("3u8hJUVTA4jH1wYAyUur7FFZVQ8H635K3tSHHF4ssjQ5").unwrap();
+
+            println!("Decode the VAA");
+            let vaa_bytes: Vec<u8> = base64::decode(vaa)?;
+            let vaa = VAA::from_bytes(vaa_bytes.clone())?;
+
+            println!("Transfer money to two accounts: signature_set and posted_vaa");
```
Comment (guibescos) [inline target_chains/solana/cli/src/main.rs:56]: This step (lines 56 to 79) is only needed on Pythnet so you can delete it

File: target_chains/solana/cli/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (guibescos) [reply target_chains/solana/cli/Cargo.toml:2]: I think you can also use that to iterate fast

**Review by guibescos — CHANGES_REQUESTED** [review]
Looks good to me, please address feedback

File: target_chains/solana/Makefile (hunk 1 above)
Comment (yhzhang0128) [reply target_chains/solana/Makefile]: Great to know! I have modified the code using `cargo run`.

File: target_chains/solana/programs/solana-receiver/Cargo.toml (hunk 3 above)
Comment (yhzhang0128) [reply target_chains/solana/programs/solana-receiver/Cargo.toml]: Reasonable. Fixed.

File: target_chains/solana/cli/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yhzhang0128) [reply target_chains/solana/cli/Cargo.toml]: Good point. I am working on it and should have it ready later this week. Will ping you on slack when I finish.

**Review by guibescos — APPROVED** [review]
Thank you!

**Pull request merged**